pcsx2/GameDatabase.cpp

```diff
@@ -14,63 +14,187 @@
+		{
+			bool fixValidated = false;
+			for (GamefixId id = GamefixId_FIRST; id < pxEnumEnd; ++id)
+			{
+				std::string validFix = wxString(EnumToString(id)).Append(L"Hack").ToStdString();
+				if (validFix == fix)
+				{
+					fixValidated = true;
+					break;
+				}
+			}
+			if (fixValidated)
+			{
+				gameEntry.gameFixes.push_back(fix);
+			}
+			else
+			{
+				Console.Error(fmt::format("[GameDB] Invalid gamefix: '{}', specified for serial: '{}'. Dropping!", fix, serial));
+			}
 		}
-	}
-	return false;
-}
 
-// Gets a string representation of the 'value' for the given key
-wxString Game_Data::getString(const wxString& key) const {
-	for (auto it = kList.begin(); it != kList.end(); ++it) {
-		if (it->CompareKey(key)) {
-			return it->value;
+		if (YAML::Node speedHacksNode = node["speedHacks"])
+		{
+			for (YAML::const_iterator entry = speedHacksNode.begin(); entry != speedHacksNode.end(); entry++)
+			{
+				// Validate speedhacks, invalid ones will be skipped!
+				std::string speedHack = entry->first.as<std::string>();
+
+				// NOTE - currently only 1 speedhack!
+				if (speedHack != "mvuFlagSpeedHack")
+				{
+					Console.Error(fmt::format("[GameDB] Invalid speedhack: '{}', specified for serial: '{}'. Dropping!", speedHack, serial));
+					continue;
+				}
+
+				gameEntry.speedHacks[speedHack] = entry->second.as<int>();
+			}
+		}
+
+		gameEntry.memcardFilters = safeGetStringList(node, "memcardFilters");
+
+		if (YAML::Node patches = node["patches"])
+		{
+			for (YAML::const_iterator entry = patches.begin(); entry != patches.end(); entry++)
+			{
+				std::string crc = entry->first.as<std::string>();
+				YAML::Node patchNode = entry->second;
+
+				GameDatabaseSchema::Patch patchCol;
+
+				patchCol.author = safeGetString(patchNode, "author");
+				patchCol.patchLines = safeGetMultilineString(patchNode, "content");
+				gameEntry.patches[crc] = patchCol;
+			}
 		}
 	}
-	return wxString();
+	catch (YAML::RepresentationException e)
+	{
+		Console.Error(fmt::format("[GameDB] Invalid GameDB syntax detected on serial: '{}'.  Error Details - {}", serial, e.msg));
+		gameEntry.isValid = false;
+	}
+	return gameEntry;
 }
 
-void Game_Data::writeString(const wxString& key, const wxString& value) {
-	for (auto it = kList.begin(); it != kList.end(); ++it) {
-		if (it->CompareKey(key)) {
-			if( value.IsEmpty() )
-				kList.erase(it);
-			else
-				it->value = value;
-			return;
+GameDatabaseSchema::GameEntry YamlGameDatabaseImpl::findGame(const std::string serial)
+{
+	if (gameDb.count(serial) == 1)
+		return gameDb[serial];
+
+	GameDatabaseSchema::GameEntry entry;
+	entry.isValid = false;
+	return entry;
+}
+
+int YamlGameDatabaseImpl::numGames()
+{
+	return gameDb.size();
+}
+
+bool YamlGameDatabaseImpl::initDatabase(const std::string filePath)
+{
+	try
+	{
+		// yaml-cpp has memory leak issues if you persist and modify a YAML::Node
+		// convert to a map and throw it away instead!
+		YAML::Node data = YAML::LoadFile(filePath);
```

Comment:
Is a full absolute path passed there? If so, this will break if the path to PCSX2 contains Unicode characters (on Windows).